runtime/new-recipe.js

```diff
@@ -548,21 +551,21 @@ class Walker extends Strategizer.Walker {
     for (var particle of recipe.particles) {
       if (this.onParticle) {
         var result = this.onParticle(recipe, particle);
-        if (result)
+        if (result && (result.constructor != Array || result.length > 0))
           updateList.push({continuation: result, context: particle});
       }
     }
     for (var viewConnection of recipe.viewConnections) {
       if (this.onViewConnection) {
         var result = this.onViewConnection(recipe, viewConnection);
-        if (result)
+        if (result && (result.constructor != Array || result.length > 0))
```

Comment:
it allows non arrays (result.constructor != Array) OR non empty arrays (result.length > 0)

this is not very readable, i'm not sure how to improve it, maybe move this expression to a  helper method:
if (!isEmpty(result)) {
...
}
where
isEmpty(result) {
  return result && (result.constructor != Array || result.length > 0);
}